SYCL/Plugin/level-zero-usm-capabilities.cpp

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: gpu, level_zero, level_zero_dev_kit
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
+// RUN: GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+// Check for queries of USM capabilities.
+// All supported L0 devices have these capabilities currently:
+//
+// CHECK:  usm_host_allocations: 1
+// CHECK:  usm_device_allocations: 1
+// CHECK:  usm_shared_allocations: 1
+// CHECK:  usm_system_allocations: 0
+// CHECK:  usm_atomic_host_allocations: 0
+// CHECK:  usm_atomic_shared_allocations: 0

```

Comment:
The ones reporting `false` (or `0` here), are they expected to stay that way or would we start seeing failures here when we have devices that start supporting these? Also, if it happens that some start supporting these aspects, would we expect other L0 devices to do the same (i.e. switching them to `1` here) or would the cases need to be removed?

Reply:
Yes, it is the expectations that the test start to fail, when/if there are L0 devices that support atomic concurrent access. At that point we will need to re-design the test in a way that it checks different things for different architectures. I'd like to keep it simple until then.

Reply:
Sounds reasonable. Thanks for clarifying.